pkg/actors/actors.go

```diff
@@ -940,6 +955,12 @@ func (a *actorsRuntime) CreateReminder(ctx context.Context, req *CreateReminderR
 		}
 	}
 
+	// Store the reminder in active reminders list
```

Comment:
Thanks for this change. but it will not resolve race condition entirely. Let's assume we have 4 reminder creation requests to create the same reminder.

1. Request 1 creates the reminder.
2. sleep 2 second
3. Request 2 and 3 create the same reminder in parallel. 
4. sleep 2 second
5. Request 4 creates the same reminder 

Request 2 and 3 could get the same stop channel and close it. then the stop channel of one of the requests can be overwritten. then request 4 can not stop both go routines, so that we have active two reminders.

I think we need creation reminder lock to resolve race condition issue completely.

Reply:
you are right